### PR TITLE
Add localisation and improve some PHP code

### LINKED ISSUE #73: Add english localisation
It could be useful to add functionality to switch localisations and add an english localisation

### PR DESCRIPTION
**Pull request by am9zZWY**
Closes #73

### CODE REVIEW

**Comment by am9zZWY** [comment]
I need help with the translation, @PhictionalOne

**Comment by am9zZWY** [comment]
@PhictionalOne @Pfonks

**Comment by am9zZWY** [comment]
image

**Comment by Pfonks** [comment]
Nice, thank you. Looks pretty good, the only thing missing that I have seen while quickly skimming the changes is that the event times are still displaying "ab" when the  locale is set to english. Potentially one could also switch to 12-hour time once english locale is selected but I don't see that as a critical part.

I'm going to review your contribution in depth once I get the time but so far I think this is pretty much ready to be merged.

**Comment by am9zZWY** [comment]
> Nice, thank you. Looks pretty good, the only thing missing that I have seen while quickly skimming the changes is that the event times are still displaying "ab" when the locale is set to english. Potentially one could also switch to 12-hour time once english locale is selected but I don't see that as a critical part.
> 
> I'm going to review your contribution in depth once I get the time but so far I think this is pretty much ready to be merged.

Yeah, I see that. I suggest to completely drop the “ab” and use the UTS time already included in the event data and just add a param if it should be on time or from that time

**Comment by am9zZWY** [comment]
image

File: localisation/en.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "index_deletedData": "Data successfully deleted from browser",
+  "kt_name": "Pub crawl",
+  "kt_text": "The first-year pub crawl<br> Tübingen is littered with small pubs and bars that have a significant impact on nightlife. In order to let the stress of the information-filled day sink in a bit, we invite you to an extensive pub crawl, where we will visit the various pubs in Tübingen's old town in small groups. Please bring enough cash with you, as you can only pay with EC cards in a few bars in Tübingen! - Volksbanken and Sparkassen can be found in the city if needed.  Register with your details below to get exact information about the meeting point and your group.",
+  "ry_name": "City rally",
+  "ry_text": "The first-year city rally<br> We'll have you and your fellow students compete against each other in teams. You will get to know interesting, beautiful and disturbing corners of Tübingen, hopefully improve your orientation in your new home and make new contacts.<br> Sign up with your details below to get exact information about the meeting point and your group.<br>",
+  "kasten_name": "Box race",
```
Comment (PhictionalOne) [inline localisation/en.json:21]: _Crate Run_ or _Crate race_ might be a more fitting translation.

File: localisation/en.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  "film_text": "The first-year film evening<br> On the evening of the first day of lectures, we will hold our film evening and we cordially invite you to join us!<br> Watch a film with us and relax after your first day of student life. Bring your new friends and your favourite snacks and let's go!<br> Drinks will be provided for a small donation.<br>",
+  "sp1_name": "Games night",
+  "sp1_text": "The first-year games evening<br> We would like to invite you to a games evening with good company and a relaxed atmosphere on the sand. The student council will provide some games and drinks (for a small fee). Of course, we would be very happy if you bring your own games, although our collection is already considerable! In order to be able to plan better and to send you a small reminder email, we ask you to register.",
+  "gr_name": "Barbecue",
+  "gr_text": "The first-year barbecue<br> You don't feel like cooking? Then you've come to the right place! The student council will barbecue with you in a convivial atmosphere. Bring whatever you need for the barbecue, our gas grill is waiting for you. Please also bring your own cutlery and crockery!",
+  "huette_name": "Freshers' hut",
```
Comment (PhictionalOne) [inline localisation/en.json:31]: _Cabin_ would sound a bit more propper. _Hut_ reminds me of something impromptu

File: index.php [hunk 3]
```diff
@@ -10,42 +15,60 @@
...
-                echo "     <p class='date'>{$date}</p>";
-                echo "  </div>";
-                echo "</a>";
-            }
-        }
-?>
+    <div class="container">
+        <label>
+            <select id="lang-selection">
+                <option value='de' <?= $localizer->getLang() === 'de' ? 'selected' : '' ?>>Deutsch</option>
```
Comment (PhictionalOne) [inline index.php:31]: The unicode symbols for :de: and :gb: would make it a bit more "engaging"

**Review by PhictionalOne — COMMENTED** [review]
thank you a lot for approaching this!

One thing I am unsure about is `endUTS` most of the events don't have a dedicated end time and are more _open-ended_. In previous years we had some complains that our very coarse duration approximations were not representative. I think it is counter productive to introduce a "set in stone" end time when the entries in our calendar are also just eye-balled

File: localisation/en.json (hunk 1 above)
Comment (PhictionalOne) [reply localisation/en.json]: holy f*** are you fast!

**Comment by am9zZWY** [comment]
> thank you a lot for approaching this!
> 
> One thing I am unsure about is `endUTS` most of the events don't have a dedicated end time and are more _open-ended_. In previous years we had some complains that our very coarse duration approximations were not representative. I think it is counter productive to introduce a "set in stone" end time when the entries in our calendar are also just eye-balled

I understand that. Therefore, I changed it now to ignore the exact time and only show the date

**Pull request merged**